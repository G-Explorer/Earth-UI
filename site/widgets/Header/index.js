import React from 'react'
import { Link, IndexLink } from 'react-router'
import Button from 'earth-ui/lib/Button'
import { Row, Col } from 'earth-ui/lib/Layout'
import { Dropdown, DropdownToggle, DropdownMenu } from 'earth-ui/lib/Dropdown'
import Icon from 'earth-ui/lib/Icon'
import pkg from '../../../package.json'
import './index.less'

const renderNav = () => {
  return (
    <ul>
      <li>
        <IndexLink to="/" activeClassName="active">首页</IndexLink>
      </li>
      <li>
        <Link to="/guide" activeClassName="active">指南</Link>
      </li>
      {/* <li> */}
      {/* <Link to="/design" activeClassName="active">设计</Link> */}
      {/* </li> */}
      <li>
        <Link to="/components" activeClassName="active">组件</Link>
      </li>
      <li>
        <Link to="/changelog" activeClassName="active">更新日志</Link>
      </li>
    </ul>
  )
}

class Header extends React.Component {
  render () {
    return (
      <Row className="header" fluid>
        <Col>
          <Link to="/" className="header__logo">
            <img src="../../img/earth_text_white.png" className="header__logo-text" />
            {/* <svg dangerouslySetInnerHTML={{__html: '<use xlink:href="#logo"></use>'}} /> */}
            <sub>v{`${pkg.version}`}</sub>
          </Link>
        </Col>
        <Col className="header__nav">
          {renderNav()}
        </Col>
        <Col right>
          <Dropdown ref="dropdown">
            <DropdownToggle className="header__nav-toggle">
              <Button icon="bars" transparent />
            </DropdownToggle>
            <DropdownMenu
              onClick={() => this.refs.dropdown.close()}
              className="header__nav-toggle-popover"
              align="middle"
            >
              {renderNav()}
            </DropdownMenu>
          </Dropdown>
          <a
            href="https://github.com/G-Explorer/earth-ui"
            target="_blank"
            className="header__github"
          >
            <Icon type="github" />
          </a>
        </Col>
      </Row>
    )
  }
}

export default Header
