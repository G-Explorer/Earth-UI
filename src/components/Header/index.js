import cx from 'classnames'
import PropTypes from 'prop-types'
import React from 'react'
import './index.less'

class Header extends React.Component {
  render () {
    const { className, children, icon, title, color, ...other } = this.props
    return (
      <header
        className={cx(
          `${prefixCls}-header`,
          `${prefixCls}-header--${color}`,
          {
            [`${prefixCls}-bg-colorful`]: color !== 'white'
          },
          className
        )}
        {...other}
      >
        <ul className={`${prefixCls}-header__bars`}>
          <li className={`${prefixCls}-header__bars-tool`}>
            <menu className={`${prefixCls}-header__bars-tool-item`}>
              {icon && (
                <li className={`${prefixCls}-header__bars-tool-item-icon`}>
                  <img src={icon} alt="icon" height={40} />
                </li>
              )}
              {title && (
                <li className={`${prefixCls}-header__bars-tool-item-title`}>
                  <label>{title}</label>
                </li>
              )}
            </menu>
          </li>
          {children && (
            <li className={`${prefixCls}-header__bars-left`}>{children}</li>
          )}
        </ul>
      </header>
    )
  }
}

Header.propTypes = {
  className: PropTypes.string,

  children: PropTypes.node,

  /** The left icon of Header */
  icon: PropTypes.string,

  /** The title of Header */
  title: PropTypes.string,

  /** The background color of Header */
  color: PropTypes.oneOf([
    'blue',
    'red',
    'orange',
    'yellow',
    'green',
    'purple',
    'pink',
    'gray',
    'slate',
    'black',
    'white'
  ])
}

Header.defaultProps = {
  color: 'white'
}

export default Header
