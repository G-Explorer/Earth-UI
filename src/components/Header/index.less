@import '~ui-variables';
@import '~ui-mixins';

@width: 100%;
@height: @ui-unit-triple;

.@{prefix-cls}-header {
  position: relative;
  z-index: @ui-zindex-topbar;
  width: 100%;
  color: @ui-color-white;
  border-bottom: 1px solid @ui-color-gray-light;
  &--white {
    color: @ui-color-black;
    background-color: @ui-color-white;
  }
  &--blue {
    background-color: @ui-color-blue;
  }
  &--red {
    background-color: @ui-color-red;
  }
  &--orange {
    background-color: @ui-color-orange;
  }
  &--yellow {
    background-color: @ui-color-yellow;
  }
  &--green {
    background-color: @ui-color-green;
  }
  &--purple {
    background-color: @ui-color-purple;
  }
  &--pink {
    background-color: @ui-color-pink;
  }
  &--gray {
    background-color: @ui-color-gray;
  }
  &--slate {
    background-color: @ui-color-slate;
  }
  &--black {
    background-color: @ui-color-black;
  }
  .ui-mixin-transition(opacity);
  &__bars {
    display: flex;
    justify-content: space-between;
    padding-left: @ui-unit-half;
    padding-right: @ui-unit-half;
    &-tool {
      position: relative;
      user-select: none;
      height: @height - 1px;
      white-space: nowrap;
      &-item {
        position: absolute;
        font-size: 0;
        white-space: nowrap;
        margin: 0!important;
        > li {
          font-size: @ui-fontsize;
          display: inline-block;
          vertical-align: middle;
          padding: @ui-unit-half @ui-unit-quarter;
        }
        li&-icon {
          height: @height;
          max-height: @height;
          min-height: @height;
          padding-left: 0;
        }
        &-title {
          label {
            font-size: 18px;
            padding: 10px;
            padding-left: 0;
            line-height: 20px;
            display: block;
            position: relative;
            top: -1px;
            overflow: hidden;
            text-overflow: ellipsis;
          }
        }
      }
    }
    &-left {
      height: 59px;
      display: flex;
      flex-wrap: wrap;
      align-content: center;
    }
  }
}
