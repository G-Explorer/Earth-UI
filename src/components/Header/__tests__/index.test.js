import React from 'react'
import { render, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import Header from '../index'

describe('Header', () => {
  it('should render title correctly', () => {
    const wrapper = render(<Header title="Header" />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render icon correctly', () => {
    const wrapper = render(
      <Header icon="https://cosmos-x.oss-cn-hangzhou.aliyuncs.com/blue-bg-webapps-ui-icon.png" />
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('should render children correctly', () => {
    const wrapper = render(<Header>Test</Header>)
    expect(wrapper).toMatchSnapshot()
  })
  it('should mount correctly', () => {
    expect(() => renderer.create(<Header title="Header" />)).not.toThrow()
  })
  it('should render className correctly', () => {
    const wrapper = mount(<Header className="test" />)
    expect(wrapper.find('header').hasClass('test')).toBe(true)
  })
  it('should render style correctly', () => {
    const wrapper = render(<Header style={{ color: 'red' }} />)
    expect(wrapper.prop('style').color).toBe('red')
  })
  it('should render color correctly', () => {
    const wrapper = mount(<Header color="blue" />)
    expect(wrapper.find('header').hasClass('true-header--blue')).toBe(true)
  })
  it('should render Chinese characters correctly', () => {
    const wrapper = render(<Header title="测试" />)
    expect(wrapper).toMatchSnapshot()
  })
})
