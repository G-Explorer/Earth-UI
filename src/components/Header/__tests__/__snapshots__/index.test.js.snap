// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Chinese characters correctly 1`] = `
<header
  class="true-header true-header--white"
>
  <ul
    class="true-header__bars"
  >
    <li
      class="true-header__bars-tool"
    >
      <menu
        class="true-header__bars-tool-item"
      >
        <li
          class="true-header__bars-tool-item-icon"
        >
          <img
            alt="icon"
            height="40"
            width="40"
          />
        </li>
        <li
          class="true-header__bars-tool-item-title"
        >
          <label>
            测试
          </label>
        </li>
      </menu>
    </li>
  </ul>
</header>
`;

exports[`Header should render correctly 1`] = `
<header
  class="true-header true-header--white"
>
  <ul
    class="true-header__bars"
  >
    <li
      class="true-header__bars-tool"
    >
      <menu
        class="true-header__bars-tool-item"
      >
        <li
          class="true-header__bars-tool-item-icon"
        >
          <img
            alt="icon"
            height="40"
            width="40"
          />
        </li>
        <li
          class="true-header__bars-tool-item-title"
        >
          <label>
            Header
          </label>
        </li>
      </menu>
    </li>
  </ul>
</header>
`;
